media/js/firefox/sticky-promo.js

```diff
@@ -43,20 +69,39 @@
         };
 
         StickyPromo.show = function (){
-        // Open promo
-            Mzp.StickyPromo.open();
+
+            function openOnScroll(){
+                // Open promo
+                Mzp.StickyPromo.open();
+                document.removeEventListener('scroll', openOnScroll, false);
+            }
+
+            document.addEventListener('scroll', openOnScroll, false);
+
+            var checkForFooterOnScroll = debounce(function() {
+                // If the footer is in the viewport, fade out the promo.
+                // Animate it back in when the footer leaves the viewport
+                // if the user did not dismiss it.
+                if (isInViewport(footer)) {
+                    promo.classList.replace('mzp-a-slide-in', 'mzp-a-fade-out');
+                } else if (!promo.classList.contains('user-dismiss')) {
+                    promo.classList.replace('mzp-a-fade-out', 'mzp-a-slide-in');
+                }
+            }, 100);
 
             // Add modifier class to the footer to make sure the language selection drop-down is not obscured by the sticky promo
             var footer = document.querySelector('.c-footer');
             if (footer) {
                 footer.classList.add('is-intersecting-sticky-overlay');
```

Comment:
Do we need this on the footer still if we're hiding the promo on scroll? Could we remove the line of CSS related to this also? https://github.com/mozilla/bedrock/blob/master/media/css/protocol/components/_footer.scss#L7-L11

Reply:
Great call. I'll remove this hotfix too.